dmutils/email/tokens.py

```diff
@@ -100,13 +100,24 @@ def decode_password_reset_token(token, data_api_client):
     try:
         decoded, token_timestamp = decode_token(
             token,
-            current_app.config["SECRET_KEY"],
+            current_app.config["SHARED_EMAIL_KEY"],
             current_app.config["RESET_PASSWORD_SALT"],
             ONE_DAY_IN_SECONDS,
         )
     except fernet.InvalidToken as e:
-        current_app.logger.info("Error changing password: {error}", extra={'error': six.text_type(e)})
-        return {'error': 'token_invalid'}
+        try:
+            # We used to issue tokens with `current_app.config["SECRET_KEY"]` as our key, so we maintain backwards
+            # compatability with them for a short period (1 day) to allow for both type of tokens to be valid until the
+            # older ones have expired and can remove the support for tokens using `current_app.config["SECRET_KEY"]`
+            decoded, token_timestamp = decode_token(
+                token,
+                current_app.config["SECRET_KEY"],
+                current_app.config["RESET_PASSWORD_SALT"],
+                ONE_DAY_IN_SECONDS,
+            )
+        except fernet.InvalidToken as e2:
```

Comment:
Could drop the `as e` from line 107, then this only needs to be `as e`. Although could leave till this gets cleand up after the transition.

Reply:
Just dropped it as it broke flake8 (strangely only in python 2 and not in python 3)

Reply:
nice